### ISSUE TITLE
Feature request: include the calculated errors of the exoplanet's period & transit depth

### ISSUE THREAD

**Issue by vvilhelmus** [lightkurve/lightkurve#1101]
#### Problem description

The .show_properties() of a periodogram does not include errors on attributes like period_at_max_power & depth_at_max_power

#### Expected behavior

When a value for a period or transit depth is obtained; some standardised calculated error or (even better) an upper & lower error should be given as well please.

#### Environment

-  lightkurve version (e.g. 1.0b6): 2.0.10

**Comment by orionlee** [comment]
If you're referring to `BoxLeastSquaresPeriodogram`, would `compute_stats()` address your need?


https://github.com/lightkurve/lightkurve/blob/81bc8697480afd371e01a9c526487f25e414ba7d/src/lightkurve/periodogram.py#L1186-L1189

It is basically a wrapper over the underlying astropy method:

https://docs.astropy.org/en/v4.1/api/astropy.timeseries.BoxLeastSquares.html#astropy.timeseries.BoxLeastSquares.compute_stats

**Comment by vvilhelmus** [comment]
Oh wow, thank you so much for this information, Sam @orionlee ! :) Now I feel silly for not knowing this... 
Upon closer inspection the .compute_stats() function is nowhere to be found in the API:
http://docs.lightkurve.org/reference/api/lightkurve.periodogram.Periodogram.show_properties.html
Other functions, which I have encountered in the tutorials, seem to be missing in the API as well, like the .get_transit_model() function for example. Could these missing functions perhaps be added to the API and/or tutorials, so newbie users like me can learn about them please?
But this .compute_stats() function does indeed give the error of the transit depth, which solves half of my request. :)
Is there also some hidden function, which can give the error in the exoplanet's period?

**Comment by orionlee** [comment]
1. `compute_stats()` missing in apidoc is certainly an issue. Maybe you can open a PR to fix it? 

2. For error in exoplanet's period from BLS: I am not an expert on BLS so I can't speak with surety, but I think the way BLS works is not equipped with providing standard errors for identified periods. The power you get back in the Periodogram does give you a measure on how well a epoch / particular period describes the data. You have control on the fit criteria with `objective`. See:

https://docs.astropy.org/en/latest/api/astropy.timeseries.BoxLeastSquares.html#astropy.timeseries.BoxLeastSquares.power

**Comment by vvilhelmus** [comment]
Assuming PR is Pull Request I am going to create one, sorry, I am new to this sort of thing.

About the error in the exoplanet's period; in the Periodogram can't one just take the half width at half maximum BLS Power of the period-peak and simply take this value as the error? Or when the peak is asymmetrical make those different values the lower & upper error of the period? My apologies for the million questions, I am just curious. I think I have seen the half width at half maximum method being used on something else in one of the tutorials, which made me wonder why it isn't being used on the period of an exoplanet.

**Comment by orionlee** [comment]
Yes, PR does mean pull requests - I apologize for using acronyms without thinking. If you're interested in contributing, see http://docs.lightkurve.org/about/contributing.html

---

> can't one just take the half width at half maximum BLS Power of the period-peak and simply take this value as the error? 

I'm afraid I am not knowledgeable enough to answer it.

**Comment by vvilhelmus** [comment]
Thank you for replying to my comment. which I had deleted, because I started to figure out what Pull Requests actually entail.
I am indeed very interested in contributing, but I have to first finish writing up my M.Sc. project during the Summer, before I can start writing any API documentation. No promises, though I think I will give it a shot at contributing to Lightkurve in the early Autumn this year. I think it is awesome you provide this application for free to the public and give citizen scientists an opportunity to do their own research this way! :+1: :)
